adonis-typings/model.ts

```diff
@@ -252,6 +252,22 @@ declare module '@ioc:Adonis/Lucid/Orm' {
    */
   export type DateTimeColumnDecorator = DateColumnDecorator
 
+  export type JsonColumnOptions<JsonType = any> = Partial<
+    ColumnOptions & {
+      replacer: (key: string, value: any) => any
+      space: string | number
+      reviver: (key: string, value: any) => JsonType
```

Comment:
I will skip these options, because if someone needs that level of customization, maybe they can self implement `prepare` and `consume` methods.

Reply:
Well, I'm not sure if my opinion will count 😄 But I think having those options makes it easy for anyone who wants such level of customisation. 

The Adonisjs framework is a super flexible and customisation framework with tons of functionalities an average user does not use or may not need for a simple project. But those functionalities are there so that the power users do not have to hack around to much to achieve what they want to do.

Reply:
Additionally, if the `prepare` and `consume` methods are overridden in an attempt to customise the stringification and parsing process, then it defeats the purpose of using this decorator in the first place as the decorator's entire functionalities will be overridden.

Reply:
I agree with @ndianabasi, I have a ```json``` decorator implemented by myself, and not having thought of the idea of receiving replacer/reviver as a parameter, I had to practically reimplement the decorator to use them when needed.

Reply:
Thumb down for these options. If your `replacer` and `reviver` don't match, you end up with broken data after several read/write cycles. It is too easy to make mistakes. For `space` option, who really care how many spaces are stored in MySQL?